extractor/main.go

```diff
@@ -0,0 +1,307 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/gnolang/gno/gno.land/pkg/sdk/vm"
+	"github.com/gnolang/gno/tm2/pkg/amino"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/peterbourgon/ff/v3/ffcli"
+	"golang.org/x/sync/errgroup"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// Define constants
+const (
+	packageMetadataFile = "pkg_metadata.json"
+)
+
+// Define errors
+var (
+	errInvalidFileType  = errors.New("invalid file type specified")
+	errInvalidSourceDir = errors.New("invalid source directory")
+	errInvalidOutputDir = errors.New("invalid output directory")
+)
+
+// Define extractor config
+type extractorCfg struct {
+	fileType  string
+	sourceDir string
+	outputDir string
+}
+
+func main() {
+	var (
+		cfg = &extractorCfg{}
+		fs  = flag.NewFlagSet("root", flag.ExitOnError)
+	)
+
+	// Register the flags
+	cfg.registerFlags(fs)
+
+	// Create the command
+	cmd := &ffcli.Command{
+		ShortUsage: "[flags] [<arg>...]",
```

Comment:
okay, so we don't actually accept any argument as they are all flags

```suggestion
		ShortUsage: "[flags]",
```